controller/src/main/java/org/jboss/as/controller/operations/common/CountDownLatch.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2018, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.controller.operations.common;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Enhanced CountDownLatch providing uninterruptible variants of await() methods.
+ *
+ * @author <a href="mailto:ropalka@redhat.com">Richard Opalka</a>
+ */
+final class CountDownLatch extends java.util.concurrent.CountDownLatch {
```

Comment:
Yes, all three enhanced implementations of CountDownLatch are identical @jmesnil .
I provided 3 of them in 3 different packages because it's not final solution yet
and I didn't want to make these classes public (to don't become part of public API).
If @dmlloyd will agree to have such class in wildfly-common I would consider it a solution
I'm comfortable with. Otherwise I prefer 3 package private implementations for now.

Reply:
I did not include this class because it doesn't introduce any appreciable improvement over having static methods which do uninterruptible await given a plain j.u.c CountDownLatch.

You could put one public class inside a private package which does this to avoid duplication.

Reply:
(A private package is defined as _private or a package inside of a module whose jboss.api property is "private").

Reply:
Getting rid of duplication is important because metaspace is always at a premium.  Even better if these methods could be made static and included an into existing class.

Reply:
I like the private API proposal @dmlloyd , reworking ...